src/tools/cephfs_mirror/FSMirror.h

```diff
@@ -91,6 +91,14 @@ class FSMirror {
     return m_addrs;
   }
 
+  bool get_init_finished() {
```

Comment:
@joscollin Now relooking at this and the test runs, I think a more appropriate fix would be what you proposed earlier to conditionally check `m_instance_watcher` and `m_mirror_watcher` in `FSMirror::is_failed()` since this check is still racy.

Please push with that fix and I'll pick this up on priority.

Reply:
oh, Let me check my backup branches if that fix is readily available.

Reply:
@vshankar 
[This](https://github.com/ceph/ceph/pull/56148/commits/5b8b9c2f7738e0341807c1e0971c3dec90ac5491) is the one you mentioned right? It is pushed to this PR